tests/conftest.py

```diff
@@ -23,24 +24,32 @@ def pytest_collection_modifyitems(config, items):
             item.add_marker(skip_slow)
 
 
-@pytest.fixture(scope="function")
-def base_config():
-    config = build_simulation_configuration()
+@pytest.fixture(scope="session")
+def base_config_factory() -> Callable[[], ConfigTree]:
+    def _base_config() -> ConfigTree:
```

Comment:
Why the misdirection?

Reply:
I think you mean "indirection", and this is because I have module scoped fixtures that use this fixture, and you can't pass a fixture with a smaller scope as an argument to a fixture with a larger scope. From my reading, it seems fixtures is the correct way to handle this.